common/src/main/scala/explore/common/RemoteSyncUndoableF.scala

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.common
+
+import cats.MonadError
+import crystal.react.View
+import crystal.react.implicits._
+import explore.undo.UndoContext
+import japgolly.scalajs.react.Reusability
+import japgolly.scalajs.react.callback.Callback
+import japgolly.scalajs.react.util.Effect
+import monocle.Lens
+import monocle.Optional
+import monocle.Prism
+import org.typelevel.log4cats.Logger
+
+/**
+ * Provides a `View` into a model with the ability to perform changes remotely whenever the `View`
+ * is modified. Additionally, operations are undoable.
+ *
+ * `A` is the model type, whereas `T` is the type of a delta structure to be sent to the remote
+ * server with only the necessary changes. For example, it can be one of the `*Input` types of the
+ * API.
+ *
+ * We can `zoom` into a property of the model while providing a parallel way to zoom into the delta
+ * structure. Potential modifications to the the delta structure are stored in a coyoneda-like
+ * fashion, but are not executed until the model is modified.
+ *
+ * The idea is to keep code to edit the model as simple as possible, while providing a way to
+ * compose components that edit different parts of the same model object in an undo context.
+ */
+trait RemoteSyncUndoableF[F[_], B, S] { self =>
```

Comment:
I really struggled with finding a name for this. I'm open to suggestions.

Reply:
This is a neat way to bundle the updates/undo. The naming is hard. I haven't been able to come up with anything better.